### DISCUSSION TITLE
ChatGPT vs GPT-3 - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by adamcogan** [SSWConsulting/SSW.Rules.Content#4947] [General]
Lets talk about these rules
https://ssw.com.au/rules/rules-to-better-gpt-3/

**Comment by tiagov8** [comment]
My feedback:

- Don't think the rule category should specifically mention "GPT-3"  
   - It also have content on ChatGPT
   - The version number is likely to change often, causing us unnecessary maintenance 
   - Maybe **Rules to Better GPT (Generative Pretrained Transformer)** would be more accurate
- I don't think we should have the first 2 rules "What is..." 
   - This information is everywhere already... we are just replicating content. We don't do that, especially for 3rd party products that describe themselves in a much better way
   - We could incorporate whatever we have extra to say about these products on the 3rd rule "[When should you use ChatGPT vs GPT-3?](https://ssw.com.au/rules/chatgpt-vs-gpt-3/)"

- Many rules titles are inconsistent...should start with "Do you..." 
- Not a big fan of adding emojis to the rule title... [Do you use ChatGPT to solve errors? 🐛⚠️](https://ssw.com.au/rules/chatgpt-can-fix-errors/)

**Reply by amankumarrr** [reply]
ChatGPT is an awesome tool. Since I started using it for my Angular project, I can't remember the last time I had to visit StackOverflow. I also use it to write captions, learn English grammar, and correct my sentence grammar.

**Reply by leehawkinsssw** [reply]
> Not a big fan of adding emojis to the rule title

I agree, emojis have no place in rule titles and their use on new rules like this will make those rules stand out for no good reason amongst existing SSW Rules. I suggest we remove the emojis and don't approve PRs for new rules that include them in the title.

**Reply by jaydenalchin** [reply]
Exciting times ahead, becoming pretty clear we're entering a new age of AI.

> Don't think the rule category should specifically mention "GPT-3"
Many rules titles are inconsistent...should start with "Do you..."
Not a big fan of adding emojis to the rule title... [Do you use ChatGPT to solve errors? 🐛⚠️](https://ssw.com.au/rules/chatgpt-can-fix-errors/)

+1 to Tiago's comments!

**Comment by BrookJeynes** [comment]
I think Chat-GPT could become a valuable tool in a developers toolkit but currently it's impossible to even use it due to it's capacity limits.

**Reply by SamBWagner** [reply]
^^

**Comment by jakebayliss** [comment]
Could've possibly talked about about how to use it better - prompt engineering is a real thing and can get much much better results. Maybe we should come up with some tips and tricks for using ChatGPT/GPT-x with code. 
I think the two rules on this could be combined into 1.

**Comment by drwharris** [comment]
I dont know what it is but I really dislike the idea of GPT. Its all "This is whats been done before". That thought train ironically stifles innovation.

**Reply by samsmithssw** [reply]
I agree, we're just regurgitating old ideas and not innovating.

**Comment by apforsyth** [comment]
Wow, I love that it can provide an explanation how a piece of code works, in standard English (or any language I presume).  Amazing!

But how accurate is it?

**Reply by Hona** [reply]
Anything with dependencies it really isnt great at. I pasted in a whole file from my personal project. It has no clue. However, it can infer what you can infer from reading a file :)

**Reply by leehawkinsssw** [reply]
Did it really "provide an explanation how a piece of code works", though? It did a good job of talking through the **logic** of the code, but I was none the wiser as to what the **intent** of that piece of code was at the end of its explanation. If it really understood the code, it would be able to say something like "this function is designed to check whether a text string contains only alphanumeric characters"

**Comment by AlexRothwell** [comment]
I think that it's great that ML models like these are now being made widely accessible to the public where they were previously difficult to use and so necessarily restricted to more technical users. This will absolutely help discover new and interesting applications and areas to investigate as the technologies continue to mature. However, I don't think the current iterations are ready for serious production use as there are still serious concerns around response accuracy

That being said, with these systems being so trendy at the moment, I think it's great to start building up a collection of Rules from our learnings. These could include useful use cases and potential applications but I feel like we must ensure that we highlight the pitfalls and things to consider as part of any of these

**Comment by ozdemir-mehmet** [comment]
ChatGPT isn't ready for internet scale

<img width="602" alt="image" src="https://user-images.githubusercontent.com/37358029/217949787-7536c66a-f88b-411e-996e-728ff0ee657e.png">

**Reply by danielmackay** [reply]
ChatGPT is currently in a 'research release' to help them improve.  Open AI will introduce a paid version in future that will be more reliable and provide quicker responses.

**Reply by ozdemir-mehmet** [reply]
I wanted to use ChatGPT to generate this reply. I couldn't :(

**Comment by Levijj22** [comment]
Exciting times ahead for the world of A.I! 
Regarding the rule, I don't think it should refer to this specific GPT version. It think it should be 'Do you use GPT to solve your coding errors?'
I agree with combining the two rules also.

**Comment by wicksipedia** [comment]
Siri gave a better answer to the question "which came first, the chicken or the egg" - it replied "I checked their calendars. They both have the same birthday!"

**Comment by UlyssesMaclaren** [comment]
I wonder if this will re-invigortate the market for Bots in general. 

As a minimum, it's gotten everyone talking about chatbots again... and also seeing how they can provide a human-interaction-esque experience.

But what I want to know is can you tell it to actualy do things? i.e. more like Siri.

How long until Siri gains more of this style of communication ability?

**Comment by danielmackay** [comment]
I think ChatGPT has awesome potential.  When seeing code being pasted into a chat window and copied out again, my first thoughts are "surely there is a VS Code plugin for this...". 

Turns out there is!  This one looks promising:
![image](https://user-images.githubusercontent.com/2636640/217950623-7936856f-6feb-4e34-9a8e-a6fa125a2f88.png)
https://marketplace.visualstudio.com/items?itemName=gencay.vscode-chatgpt

That leads to the next obvious question...which AI coding assistant is better out of ChatGPT and Copilot?

**Comment by danielmackay** [comment]
I also saw this ChatGPT demo recently, which blew me away!

https://www.youtube.com/watch?v=z2CKQFi746Q

**Comment by SamBWagner** [comment]
I like AI, I like ChatGPT. It's use is WAY overstated and I think there are really good niche, small places that it's useful: Writing Tweets, PBI text, small functions, Documentation... 

Using it for more than that is usually cumbersome and the availability makes it hard -> impossible to use

**Comment by cvionnet** [comment]
ChatGPT is a fantastic tool, I use it more and more.
And it is also good to see competitors coming with Baidu and Google.

About the rules' main title, it could be more generic, like:
From:
    Rules to Better GPT-3
To:
    Rules to Better Coding with AI

**Comment by TobyChurches** [comment]
I think ChatGPT is a very interesting tool with a number of valuable use cases. When used correctly, I think that I could definitely save the developer time...

However, its availability is an issue, with the current capacity limits falling short of its popularity (leading to longer wait times 😬).

**Comment by JeanThirion** [comment]
ChatGPT is definitely an interesting tool and has the potential to start a little internet revolution.
However, when it comes to coding, I'm not convinced. Anyone that has used ChatGPT to generate some real code knows it sucks - I'm not scared for my job!

**Comment by leahy268** [comment]
Chat GPT is interesting.  And probably very useful for bots and things like that where there is a known source of information.  However as others have said it only gives back what was done before.  The only way to solve a problem is xxx. That inherently stops people from innovating new ways of doing things.

**Comment by tombui99** [comment]
ChatGPT is an awesome tool! 
I can use it to find code problems, code templates for specific tasks (much easier compared to looking up on StackOverflow), it can also write emails for me and do all sort of tricks! Just hopefully it won't replace my job one day lol.

**Reply by leehawkinsssw** [reply]
I think you're pretty safe, @tombui99 The hard part of being a developer isn't typing the code, right? The tricky part is understanding the problem you're trying to solve and then encoding a solution to that problem in whatever language you're using. While tools like this **might** do a reasonable job of the latter, their ability to meaningfully understand the problem is next to non-existent.

**Comment by stephanfakossw** [comment]
It's an interesting app. I think it's rather limited in application for general questions but it has great potential for engineering or science. 
The rules are concise. Although it might have been better to merge both rules into 1.

**Comment by KostaMadorsky** [comment]
ChatGPT is a great resource to use. 
Great that it also generates a quite detailed explanation for the code it generates, which encourages developers not just to copy-paste code but also to understand how exactly it works.
At this stage, it's important to mark the solutions that ChatGPT gives you so the model can learn from our feedback.

**Comment by AntPolkanov** [comment]
ChatGTP is a great tool for learning tools/frameworks as the answers to _"How do I ... ?"_ questions are pretty good! Won't help with constantly emerging JS frameworks though 😜

**Comment by christoment** [comment]
ChatGPT looks good, it performs well and responded with enough context to give me enough data to move on with/without using the given code.

The biggest problem I think is that the result does not (or cannot) cite any resources it use to draw the conclusion - it means that we are stuck in either to trust the answer, or need to cross check the answer. 🙂

**Comment by vladislav-kir** [comment]
Chat GPT is an interesting product, although I think the situations where you would seriously benefit from using it are quire rare. Additionally, it seems to have a very limited availability, and waiting to access your chat might now worth the time.

**Comment by Hona** [comment]
(in light of this CTF, my thoughts are the thoughts of ChatGPT)

AI has revolutionized the software development industry with its ability to improve code and fix bugs more efficiently.

OpenAI has been at the forefront of this revolution, with their announcement of ChatGPT and GPT-3, two powerful language models that offer unique features and benefits.

While ChatGPT provides a fun and interactive platform for users to chat with, GPT-3 takes things to the next level by exposing its capabilities through an API for developers to incorporate into their projects.

With the API, developers have the ability to build applications that can understand natural language and generate human-like responses. This can greatly enhance the user experience and improve overall efficiency.

The integration of ChatGPT into the API will further enhance the capabilities of OpenAI's offerings, making it a one-stop-shop for all language model needs.

As technology continues to evolve, it's exciting to think about what new advancements we'll see in the field of AI and how they will continue to change the way we work.

Chewing the Fat is a great platform for sharing ideas and insights on the latest developments in AI and software development, and I look forward to seeing what new topics will be covered in the future.

**Comment by christianmorfordwaitessw** [comment]
Looks like it could be a useful tool. The demos and examples people have tried on it look promising.
I haven't had a chance to use it due to all the over capacity issues it has

**Comment by camillars** [comment]
I'm loving using ChatGPT for marketing. AI has been trending in the marketing industry for a while now, so it's good to do some practical and easy applications in the day to day activities. I have been using it almost on a daily basis.

**Comment by Harry-Ross** [comment]
I think that ChatGPT is going to be a transformative tool for developers and others alike. I have already used it for a variety of coding-related questions and think it has helped me out immensely. Personally, there is still a fair way to go, as it can occassionally provide unreliable answers to some questions - but I'm excited to see how they can improve it.

**Comment by bahjatalaadelssw** [comment]
This might be helpful for concept generation to guide the execution of a product or ideation in general. I struggled to ask it a question that an internet search wouldn't resolve. It seems like a search complier.

**Comment by leehawkinsssw** [comment]
I agree with previous comments about removing the tight coupling to GPT-3, the rules should be more about using AI services like these and less about particular services. It's good to use specific examples, of course.

We need to be careful in saying things like ChatGPT "has been trained to understand how code functions" - it doesn't truly understand anything and often gives plausible sounding but incorrect or nonsensical answers. It currently struggles with logic and consistency. Have a look at the mess it makes of these couple of examples I tried this morning:

is 1000000 smaller than 1000050?
how many minutes would it take me to write out the digits of 10^35?

Such basic logical errors lead me to question its veracity in tackling tricky coding and, of my particular interest, testing challenges.

Another big concern for me is the potential for systems like these to magnify bad/unwanted code patterns, similarly to when CoPilot 
 was suggesting the use of code containing client-specific secrets (since fixed).

There are significant ethical concerns too. The OpenAI Chief Executive Officer, Sam Altman, suggested people “thumbs down” bad, incorrect or ethical dubious results to help the tech improve, but it's disappointing that the company is putting the responsibility on users to fix the problem. The architecture, training data and goals need to be looked at more seriously from an ethical perspective. As a case in point, this system is already coming at a cost for some of our fellow humans https://time.com/6247678/openai-chatgpt-kenya-workers/

**Comment by JackDevAU** [comment]
Firstly, I'm not a fan of the emojis in the titles of rules as others have mentioned above. They just look out of place.

Using AI will be a great help to quickly understand complex issues.... however, this does come at the downside of us relying heavily on these tools. Currently ChatGPT makes educated guesses when reading snippets of code and can often make up a lot of functions if you ask it how to do a certain thing (that it thinks should be there). So currently it is still a ways off from becoming something that anyone can reliably use for coding. For marketing, letters etc I think its an awesome tool with a lot of promise!! 

Since I could actually connect to ChatGPT today, I decided to ask it what it thought about the programming industry 

### Will ChatGPT revolutionize the programming industry

> It is possible that ChatGPT and other AI models could have a significant impact on the programming industry. AI models like ChatGPT have the potential to automate certain tasks, such as code completion, bug fixing, and code generation, which could greatly improve the efficiency and accuracy of software development.

> However, it is also important to note that AI models like ChatGPT are still in the early stages of development and have limitations. For example, they may not always produce optimal or secure code, and they may struggle to understand the intent behind certain code constructs.

> Additionally, AI models do not yet possess the creativity and problem-solving abilities of human programmers, so it is unlikely that they will completely replace human programmers in the near future.

> In conclusion, while ChatGPT and other AI models have the potential to revolutionize the programming industry, they will likely play a complementary role to human programmers, rather than replace them.

**Comment by Geordie88** [comment]
I liked the look at the bots (the side-by-side comparison especially), I haven't played with them before so this was neat :). That said the rules that are essentially just "hey try pasting this into ChatGPT feels weird to me, it's a very different UX from any of the other rules I have interacted with.

**Comment by piers-sinclair** [comment]
I'm excited to see what can be cooked up by the API, but not as excited about using ChatGPT like google, it still seems unreliable.

**Comment by AdriT9** [comment]
AI tools are good fun to play with. Within a work environment, as a non-dev, I would use ChatGPT for marketing like writing captions for social media, for example. 
A lot of people have been using & playing around with AI - creating entire stories or illustrations/ paintings of themselves. As these tools improve & evolve, there are a lot of factors to consider, like where do we draw the line...

**Comment by chrisschultzssw** [comment]
Interesting times ahead! I like that the focus of the CtF was more about troubleshooting issues, not getting AI to write your code for you - as others have said, there will be more innovation with humans writing the code.

**Comment by steven0x51** [comment]
AIs such as ChatGPT are definitely going to play an important role in the future, but at this moment they are still a plaything.

**Comment by jontygardner** [comment]
We need a rule on ChatGPT from the marketing angle:

[www.ssw.com.au/rules/do-you-use-chatgpt-for-content-creation](url)

ChatGPT is going to be very impactful for people who get 'writer's block' by generating ideas or writing sentences in mere seconds. It is going to save marketing teams and creatives hours and eventually days by generating content in seconds. It is not perfect, but so isn't anything humans write. Here are some things we can use it for:

- Script writing
- Voice-over generation: ChatGPT can be used to generate voice-over scripts for corporate films, which can be used to narrate the story and provide background information.
- Social media content

**Comment by jimmy1992abc** [comment]
I had been using ChatGPT for weeks and I quite love this tool, it helps me a lot. 
Here are the 👍 and 👎from my personal opinion.

Good:
- Help users to create writing content, especially something like marketing material, because ChatGPT can access a wide variety of material online that may have something in common.
- Help users fix the coding, excel formula, or generate an idea of how to code certain tasks. Because coding or formula are designed under a framework and ChatGPT not only copy the answer but also conclude the framework. GitHub could be also a great database to learn from.
- Good at explaining the concept of a specific term, helping users in learning something new. The more details the term is, the more information ChatGPT can provide to you.
- It creates content not only based on the question but also on the conversation, which means I can always discuss the issue more without adding more and more content to my question.

Bad:
- If the users want to compare something that has limited content online, ChatGPT can only provide you with the concept of each term and limited written content online. It can not actively do an analysing and generate something that does not have something similar existing online.
- It doesn't provide the relevant source of the answers for further information (it may be designed to be like this, and there is also another tool from OpenAI called WebGPT that especially provide this kind of information)
- When the answer is complex or hard to answer, ChatGPT would freeze without an answer. Ideally, it should provide the user that he is not able to help, then lead the user to try relevant information somewhere else or different terms.

Overall I think ChatGPT is helpful for certain tasks that are clear or under a framework. ChatGPT can actively provide some choices but can not lead you out of the puzzle if you are totally lost in direction. 

Also, I have a concern about ChatGPT, the AI is based on the feedback from the users and database, what if a BOT is created to intensively mislead ChatGPT, for example, thumb down the right answer and thumb up the answer he wants, repeat millions of times, would it change the information that everyone received?
Answers are welcomed.

**Comment by zacharykeeping** [comment]
I think this is an interesting tool and I'm excited to see how it continues to advance and adapt. I'm wary about relying on it too much in its current state as it can confidently give incorrect answers, but it seems like a good way to basically give you the gist of what you need an answer to very quickly.

**Reply by leehawkinsssw** [reply]
As a case in point about its overconfidence:

is 10^45 bigger than 10^46?
Yes, 10^45 (10 to the power of 45) is greater than 10^46 (10 to the power of 46). The value of 10^45 is approximately 10 followed by 45 zeros, and the value of 10^46 is approximately 10 followed by 46 zeros, so 10^46 is a larger number than 10^45.

how confident are you in the last answer?
As an AI language model, I am programmed to provide accurate answers. I am confident that the answer I provided regarding the comparison between 10^45 and 10^46 is correct.

**Comment by jernejk** [comment]
I'm using ChatGPT to brainstorm ideas, generate stuff like docs, emails, tweets, etc. that needs to have that machine touch that follows the rules, asking it "What do you think this rule wants from me? I just want to do that, please help me, I don't want to write all of this! What is the least path of resistance and do you see a loophole to bypass the rule entirely?" and in rare cases, I use it as part of coding.

But I generally don't use it for technical parts as the knowledge base is from 2021 and I want the best answer for 2023. It can be useful for an obsolete project where the knowledge of how to do or fix something is lost even to Google.

**Comment by Ben0189** [comment]
**From a typical user perspective:**
I utilize it to rephrase sentences and get some idea on how I can articulate some things in English

**It helps me to :**
- Improve my ability to communicate effectively.
- Improve my writing skill
- Acquire a broader vocabulary.
- Accelerate the process of learning in English
- Most significantly, enhanced my confidence in using English.

**Feedback** - As a non-native speaker of Australian English with limited exposure to the country, I frequently face challenges in vocabulary and sentence structure while speaking. I am grateful for the invention of ChatGPT, as it will be a valuable tool in addressing these weaknesses and improving my English.

**Developer perspective:**
I do utilize ChatGPT as a resource while coding from time to time.

**Pros:**
It can provide new ideas and solutions, as well as assist with basic programming problems.

**Cons:**
The code is unreliable and may not be equipped to handle complex and highly customized tasks.

**Feedback** - It is certainly a useful tool for developers to increase their productivity, but it is still not at a stage where it can fully replace major human work. Currently, it offers valuable assistance, but it is not to the point where it can be fully relied upon. As AI and humans continue to learn from each other (eg. I improve English from ChartGPT suggestion and ChatGPT learn to provide better suggestion to me later using the data from the conversation). I believe we are in a mutually beneficial situation currently. I am enthusiastic about the progress of AI technology and how it can assist us, but I do not want it to entirely replace human efforts.

*I used ChartGPT to support me in composing the paragraph above.

**Reply by leehawkinsssw** [reply]
A really interesting perspective, thanks @Ben0189 !

**Comment by KahaMason** [comment]
I like the idea for ChatGPT, I haven't used it much personally, and I see the appeal of what it provides. I think it definitely a good example for the benefit of AI in the general marketplace.

**Comment by NadeekaJK** [comment]
I'm optimistic about it but wouldn't depend on it. It's feeding off of existing info, not as innovative/judgmental/critical as the human mind.

**Comment by AshAnil** [comment]
I personally haven't used both GPT-3 and ChatGPT. After a few trials, I find ChatGPT more useful as it can provide answers to a wide range of questions on various topics, including but not limited to history, science, current events, and more.
ChatGPT can also generate text for you, such as writing a story or composing an email.

"For instance, if you want to generate human-like text or perform a diverse range of NLP tasks, GPT-3 may be a better option. If you want to have a conversational AI model that can answer questions and have text-based conversations, then ChatGPT may be the way to go." - ChatGPT

**Comment by jerryluossw** [comment]
ChatGPT is interesting, but it can't be used in China directly, so I'm looking forward to Auzre ChatGPT Service

![image](https://user-images.githubusercontent.com/68266237/217983819-514cbb54-d6f8-48a1-a958-8151b08c9107.png)

**Reply by AlvinShenSSW** [reply]
I am also aware it is not available to access from China directly.
But there are couple of ways to get around. try to google it.

**Comment by hveraus** [comment]
The beauty of this tool is that ChatGPT is a self-help tool. It can be used as a tool to shape your ideas. The analytical version of yourself, the one that does the work. For example, in software development, you ask it because you know there is an answer, but you do not want to spend the time writing syntax, so you use it as a self-assistant to write the code for you. ChatGPT is by far the best self-assistant I've ever used. Just formulate your ideas and let ChatGPT formulate them in any language you want.

**Comment by jimmidier** [comment]
5 reasons ChatGPT remains a toy (though much more knowledgable than a toy) to me
1. Data from secondary sources is nearly impossible to validate - In a sense integrating it into Bing mitigates the problem
2. Not accepting commands means it's essentially just a wikipedia - In a sense by integrating it into RPA provides the ability
3. No proactive ability means it relies on human users to teach and train it
4. ...But it always forget what it've learned once the conversation ends
5. No sense of humor... Not even a bit

**Comment by AlvinShenSSW** [comment]
ChatGPT is the future but still not mutual enough. 
I found it pretty handy to help me write articles in foreign languages. 
I can use the structure it wrote and then modify the key content by myself.

**Comment by sylhuang** [comment]
I like how powerful ChatGPT is but I'm also concerned about the accuracy of its answers. From my experience, it rarely states anything about its confidence level and while all statements seems to be in a firm tone, there could be obvious flaws or assumption that I don't know where it is basing from. This leads to the fear that if I'm asking something I have zero knowledge of, I could not spot the mistake at all and could take that information as a fact. Scary.

**Comment by SofieHong** [comment]
I think it's a great text authoring tool, especially for video scripting and optimization. However, it is not mature enough, and the output text needs further processing, such as adding emotional echo points, conveying emotions, adding punchlines, etc.

**Comment by Berber-Mama** [comment]
The video Adam shared (ChatGPT vs. GPT-3 - Which Is Better? - Q&A Comparison) made me feel creepy, from my sight, GPT-3 is more like a functional robot but ChatGPT sounds more like a natural person because it expands the topic appropriately in accordance with the human mind rather than giving direct answers to specific questions. I can't help but imagine how much more like a human mind it might grow into in the future...Amazing and incredible.

**Comment by KristenHu** [comment]
Cool tech, but it's not available in China, though I can log in to the page, it's still hard to use

**Comment by AnkyChe** [comment]
ChatGPT is interesting, but not avallable in China, so it's not convenient for me. 
From Adam's vedio, when ChatGPT answers quetiones, it gives more context which feels great!

**Comment by AttackOnMorty** [comment]
From my perspective, ChatGPT has great potential but it needs to
- Improve accuracy and credibility
- Support voice input/output. For most of the time, typing and reading suck.
- Integrate with other products (For example, integrate with code editors so it can pair programming like a real human)

I hope to see more good AI products in the wave of ChatGPT.

**Comment by johnxu16** [comment]
ChatGPT is so useful for writing structural content. I'll use it more often in my daily work.

**Comment by evecogan** [comment]
Great to understand the difference between ChatGPT and GPT-3!

**Comment by Freego1783** [comment]
Cool tech ! But maybe not mature enough for now.

**Comment by premradhakrishnan** [comment]
Many people I spoke to previously considered AI as something that had a lot of potential but not fully ready yet. ChatGPT and GPT-3 have changed this perception and boosted people's confidence in AI systems.  

TLDR: should translate to more AI projects for us.

**Comment by tkapa** [comment]
AI - Cool in concept. 

The application of AI I've seen thus far has been boring, to say the least. I'd use it more as a tool if it wasn't so annoying to do so. 

Why should I use ChatGPT, whom hired $2/h Kenyan labor to ensure that their bot wasn't racist?

**Comment by Marxoz** [comment]
ChatGPT and GPT-3 can be great to help with coding tasks. I don't know whether I will use it for more complex tasks as it is based on previous coding examples which might not be the best way to do it. But it will still be a great starting point if you are stuck

**Comment by bradystroud** [comment]
Cool concept - I won't use it much, but I can see it being integrated into many services to add a bit of AI spice 🤌🏼

**Comment by Calvin-Lieu** [comment]
Looks like it could be very helpful in the future, but for now I'll just try to fix my own code so I can more effectively reduce errors I make during the development process in the future, independent of tools like this :).

**Comment by PennyWalker** [comment]
Cool way to look at the differences in the AI experience. I think AI can be a really useful tool, but I think it needs to be used with caution. Generally, I think ChatGPT gave the better responses but they are still quite robotic. 

I have to agree a little with Andrew that it might stifle innovation because people are thinking less about what they are doing, but I would be interested to see how it develops.

**Comment by calumjs** [comment]
Good start, but I think our rules don't show the full potential of this stuff.
Let's update after my user group presentation.

**Reply by jernejk** [reply]
I agree and the thing with ChatGPT is that most of the time, the answers won't be spectacular and most use cases that we'll won't be eye popping.

It's like home automation, you might notice that someone configured their home that when TV is switch on or off, the lights are changed accordingly. No one will notice that temperature and humidity is finely controlled for maximum comfort with minimum energy consumption based on the weather outside, prioritizing rooms (over climate control entire house) and predictive scheduling.

It's these kind of things, nobody notices, that will make the biggest difference. Imagine you wouldn't ever need to write timesheets, sprint emails, meeting requests, etc, and at some point you would forget that it takes a lot of effort to do these things and that you actually had to do them (at that time, you'll call other people crazy for doing these things manually, why?). Imagine you could just say "What emails do I need to action today?" or "Are we on track with the Sprint?", "What feature took us by surprise in terms of estimates?", "What SSW rule do I need to do XYZ? Give me URL, summarize the important part for what I need to do and if I need to send an email or event, just draft it for me." where XYZ is the only part you need to fill in. (I need this 7 years ago)

In fact, spam filters, intrusion detection, Google (they'll soon use Bard, GPT-3 competitor), many creators on various platforms, etc. are already actively using GPT-3 like AI. Lots of daters get significantly higher success with help of ChatGPT! (although, it's not as effective for women as it is for men, likely due to data being skewed toward picklines for men to use) And it does work better than without. 😊

True power comes with the API usage of GPT-3 and custom training, where you get nearly for free an insanely powerful NLP (natural language processing) outperforming most NLP models. (They don't even get close, even if you use Cognitive Services)

And for creativity, AI does it similar as humans, remix the heck out of existing stuff. 🤣 (I'm building a Magic the gathering: Cloud Walker rules where the main goal is to pass AZ-204 exam by playing correct cards against an opponent, although most creativity is coming from me, ChatGPT is cracking it to 11)

**Reply by calumjs** [reply]
Good point JK, if you still know you are using AI, there is improvement to be made.
I agree with your point about how cool it is to have such a powerful model available for so cheap...
On creativity - human creativity is very hard to force, but AI creativity is easy to force - it's just often not good - so it's more like searching for gold nuggets.

**Comment by matt-goldman** [comment]
ChatGPT is cool. You still need to know how to ask the right questions and interpret the answers. It doesn't replace developers but it's a great tool to have in your kit.

**Comment by JackDuan123** [comment]
There will be a day that AI eliminate all developers!!!!!
What's the value of being a developer or even the value of being human being are under huge challange.

I want to be a basketball trainer if AI takes my Job

**Comment by DavidAbraham082** [comment]
it's fun but still has major sticking points
